### PR TITLE
claimserver

### PR DESCRIPTION
**Pull request by arnaucube**
add `claim server` module

### CODE REVIEW

File: cmd/claimserver/commands/server.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	ks, acc := cfg.LoadKeyStore()
+	client := cfg.LoadWeb3(ks, &acc)
+	storage := cfg.LoadStorage()
+	mt := cfg.LoadMerkele(storage)
+
+	rootservice := cfg.LoadRootsService(client)
+	claimservice := cfg.LoadClaimService(mt, rootservice, ks, acc)
+	adminservice := cfg.LoadAdminService(mt, rootservice, claimservice)
+
+	// Check for founds
```
Comment (ed255) [inline cmd/claimserver/commands/server.go]: typo: should be funds

File: cmd/claimserver/config.example.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+server:
+	serviceapi: 0.0.0.0:6000
```
Comment (ed255) [inline cmd/claimserver/config.example.yaml]: Inconsistent indentation.  Use 2 spaces like in the rest of this file.

File: cmd/claimserver/endpoint/admin.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		return
+	}
+	if len(m.Data) != 496/8 {
+		c.String(http.StatusBadRequest, "data smaller than 496/8")
+		return
+	}
+
+	var indexSlot [400 / 8]byte
+	var dataSlot [496 / 8]byte
+	copy(indexSlot[:], m.IndexData[:400/8])
```
Comment (ed255) [inline cmd/claimserver/endpoint/admin.go]: You are copying a string into a binary array.  Is this what you wanted?

File: cmd/claimserver/endpoint/admin.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	c.String(http.StatusOK, "imported "+strconv.Itoa(count)+" key,value entries")
+}
+
+func handleClaimsDump(c *gin.Context) {
+	r := adminservice.ClaimsDump()
+	c.JSON(http.StatusOK, r)
+}
+
+type addClaimBasicMsg struct {
+	Namespace string
```
Comment (ed255) [inline cmd/claimserver/endpoint/admin.go]: Namespace is not used

File: cmd/claimserver/endpoint/admin.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	c.String(http.StatusOK, "imported "+strconv.Itoa(count)+" key,value entries")
+}
+
+func handleClaimsDump(c *gin.Context) {
+	r := adminservice.ClaimsDump()
+	c.JSON(http.StatusOK, r)
+}
+
+type addClaimBasicMsg struct {
```
Comment (ed255) [inline cmd/claimserver/endpoint/admin.go]: This could look nicer with a json.Unmarshaler that also validates the input.  Let's talk about this later.

**Comment by ed255** [comment]
Also, I've opened this issue #87 to organize the code better.  In this commit I can see that some code is copied from other servers.

File: cmd/claimserver/commands/server.go (hunk 1 above)
Comment (arnaucube) [reply cmd/claimserver/commands/server.go]: right, thanks

File: cmd/claimserver/config.example.yaml (hunk 2 above)
Comment (arnaucube) [reply cmd/claimserver/config.example.yaml]: done

File: cmd/claimserver/endpoint/admin.go (hunk 3 above)
Comment (arnaucube) [reply cmd/claimserver/endpoint/admin.go]: solved

File: cmd/claimserver/endpoint/admin.go (hunk 4 above)
Comment (arnaucube) [reply cmd/claimserver/endpoint/admin.go]: removed

File: cmd/claimserver/endpoint/admin.go (hunk 5 above)
Comment (arnaucube) [reply cmd/claimserver/endpoint/admin.go]: agree, after this we can make a branch and abstract those parts of code for all the servers

Comment (ed255) [reply cmd/claimserver/endpoint/admin.go]: We leave this for future iterations.

**Pull request merged**